Rewrite README (#205)
seek-oss/vocab: README.md

> [!NOTE]
> Using methods discussed later we'll make sure the first language is loaded on page load.
> However, after this, changing languages may lead to a period of no translations as Vocab downloads the new language's translations.

```tsx
// src/App.tsx

import { VocabProvider } from '@vocab/react';

const App = ({ children }) => {
  return (
    <VocabProvider language="en">{children}</VocabProvider>
  );
};
```

If you need to customize the locale for your language, you can pass a `locale` prop to the `VocabProvider` component.
This tells Vocab which locale to use when formatting your translations.

```tsx
// src/App.tsx

import { VocabProvider } from '@vocab/react';

function App({ children }) {
  return (
    <VocabProvider language="myCustomLanguage" locale="en">
      {children}
    </VocabProvider>
  );
}
```

See [here][overriding the locale] for more information on how and when to use the `locale` prop.

[overriding the locale]: #overriding-the-locale

### Step 4: Create translations

A translation file is a JSON file consisting of a flat structure of keys.
Each key must contain a `message` property, and optionally a `description` property.

> Rather than creating one giant file for each language's translations, Vocab enables you to locate translations as close as possible to the components that use them.

Reviewer: Rephrasing the second half, may be subjective

```suggestion
Rather than creating one giant file for each language's translations, Vocab enables you to co-locate the translations alongside their consuming components.
```
